sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,707 @@
+"""Gaussian processes classification."""
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import warnings
+from operator import itemgetter
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+from scipy.special import erf
+
+from sklearn.base import BaseEstimator, ClassifierMixin, clone
+from sklearn.gaussian_process.kernels \
+    import RBF, CompoundKernel, ConstantKernel as C
+from sklearn.utils.validation import check_X_y, check_is_fitted, check_array
+from sklearn.utils import check_random_state
+from sklearn.preprocessing import LabelEncoder
+from sklearn.multiclass import OneVsRestClassifier, OneVsOneClassifier
+
+
+# Values required for approximating the logistic sigmoid by
+# error functions. coefs are obtained via:
+# x = np.array([0, 0.6, 2, 3.5, 4.5, np.inf])
+# b = logistic(x)
+# A = (erf(np.dot(x, self.lambdas)) + 1) / 2
+# coefs = lstsq(A, b)[0]
+LAMBDAS = np.array([0.41, 0.4, 0.37, 0.44, 0.39])[:, np.newaxis]
+COEFS = np.array([-1854.8214151, 3516.89893646, 221.29346712,
+                  128.12323805, -2010.49422654])[:, np.newaxis]
+
+
+class _BinaryGaussianProcessClassifierLaplace(BaseEstimator):
+    """Binary Gaussian process classification based on Laplace approximation.
+
+    The implementation is based on Algorithm 3.1, 3.2, and 5.1 of
+    ``Gaussian Processes for Machine Learning'' (GPML) by Rasmussen and
+    Williams.
+
+    Internally, the Laplace approximation is used for approximating the
+    non-Gaussian posterior by a Gaussian.
+
+    Currently, the implementation is restricted to using the logistic link
+    function.
+
+    Parameters
+    ----------
+    kernel : kernel object
+        The kernel specifying the covariance function of the GP. If None is
+        passed, the kernel "1.0 * RBF(1.0)" is used as default. Note that
+        the kernel's hyperparameters are optimized during fitting.
+
+    jitter : float, optional (default: 0.0)
```

Comment:
I assume this name is the standard term in the GP literature for l2 regularization? Since in the rest of sklearn these parameters are named `alpha` or `1/C`, I would consider adjusting this. Also, this seems to be redundant with composing with the constant/white noise kernel? Although it is so important that it should get its distinguished place in the model.